src/ui/opengl/castletiledmap_map.inc

```diff
@@ -1034,6 +1034,18 @@ begin
     end;
   finally FreeAndNil(I) end;
 
+  { TODO : Handle tiles which are not explicitly found in the tsx-file.
+    Hint : Tiles which have no special setting (probability equals 1.0,
+           no terrains set, no type set) are not explicitly written down
+           in the tsx file. This means they are not loaded by the iteration
+           above and consequently are not found in the Tiles list. }
+  if not (TileCount = Tiles.Count) then
+  begin
+    WritelnWarning(URL + ': Could not load all tiles of tileset');
+    WritelnWarning('Expected tiles: ' + IntToStr(TileCount) +
+      ' Loaded tiles: ' + IntToStr(Tiles.Count));
+  end;
+
   if TileCount = 0 then
```

Comment:
Question: Should we move this check later, *after* the block 

```
  if TileCount = 0 then
  begin
    Image.DetermineSize;
    TileCount := (Image.Width div TileWidth) * (Image.Height div TileHeight);
  end;
```

? This block potentially adjusts `TileCount`. So the check you propose should be after it?

Reply:
Yes, you are right. I would even propose to add a Log entry to indicate that the tsx-file is outdated if it doesn't carry a tilecount entry, as this is the case with tsx files since ver. 0.13 (2015). 

Or could you imagine any other scenario in which TileCount = 0 is possible even in recent tsx-files?